awx/api/views/__init__.py

```diff
@@ -2567,6 +2567,12 @@ def _validate_spec_data(new_spec, old_spec):
                 return Response(dict(error=_(
                     "Survey question {idx} of type {survey_item[type]} must specify choices.".format(**context)
                 )))
+            elif qtype in ['multiplechoice', 'multiselect'] and 'choices' in survey_item:
```

Comment:
I'd suggest using just one overall conditional for the `qtype in ['multiplechoice', 'multiselect']` part used here, below, and above, and then have nested under it a new if/else block or blocks for these three conditionals.

Reply:
I would prefer that as well, but this is the existing structure for above checks in the code. Maybe time for an update of those as well?

Reply:
Yeah, I'd vote that you go ahead and refactor the prior check as well. This is a very minor re-organization. If it's a choice field, then you go into the conditional block. Then you do different actions depending on whether choices are provided. You could even do it with only 1 extra conditional, `if 'choices' not in survey_item`, because it will `return`, the execution will not continue after that.